### PR TITLE
Fix OADP-5183: Ensure containers are all using FallbackToLogsOnError in terminationMessagePolicy

### PR DESCRIPTION
**Pull request by shubham-pampattiwar**
## Why the changes were made

Ensure containers are all using FallbackToLogsOnError in terminationMessagePolicy

Related JIRA: https://issues.redhat.com/browse/OADP-5183

## How to test the changes made

- Install Operator and create basic DPA deploying Velero, Node-Agent and NAC Pods, alongside the OADP Operator pod in install NS
- Use the Certsuite tool to test the changes: https://github.com/redhat-best-practices-for-k8s/certsuite 
- You need to check wether the PR changes pass for the observability test: `observability-termination-policy`
- Follow the setup docs: https://redhat-best-practices-for-k8s.github.io/certsuite/configuration/ 
- And then run the relevant test for our pods, For this PR to be tested, run: 
```
spampatt@spampatt-mac certsuite [main]$ ./certsuite run -l "observability-termination-policy" -c certsuite_config.yml
```
- You can see results under the `results/` (logs, results html etc)
- Should look like:
Screenshot 2024-11-13 at 5 07 05 PM

### CODE REVIEW

File: controllers/velero.go [hunk 1]
```diff
@@ -425,7 +425,7 @@ func (r *DPAReconciler) appendPluginSpecificSpecs(veleroDeployment *appsv1.Deplo
 					ImagePullPolicy:          imagePullPolicy,
 					Resources:                init_container_resources,
 					TerminationMessagePath:   "/dev/termination-log",
-					TerminationMessagePolicy: "File",
+					TerminationMessagePolicy: corev1.TerminationMessageFallbackToLogsOnError,
```
Comment (mateusoliveira43) [inline controllers/velero.go:428]: here would not be better to use `setContainerDefaults` function from controllers/common.go?

File: controllers/nonadmin_controller.go [hunk 2]
```diff
@@ -176,6 +176,7 @@ func ensureRequiredSpecs(deploymentObject *appsv1.Deployment, image string, imag
 				nonAdminContainer.Image = image
 				nonAdminContainer.ImagePullPolicy = imagePullPolicy
 				nonAdminContainer.Env = []corev1.EnvVar{namespaceEnvVar}
+				nonAdminContainer.TerminationMessagePolicy = corev1.TerminationMessageFallbackToLogsOnError
```
Comment (mateusoliveira43) [inline controllers/nonadmin_controller.go:179]: need to apply this also to `if len(deploymentObject.Spec.Template.Spec.Containers) == 0` case

File: controllers/velero.go (hunk 1 above)
Comment (shubham-pampattiwar) [reply controllers/velero.go:428]: We use the `setContainerDefaults` function for velero and node-agent containers, here are we setting values for plugin init containers. so I think current way of things is fine for now.

File: controllers/nonadmin_controller.go (hunk 2 above)
Comment (shubham-pampattiwar) [reply controllers/nonadmin_controller.go:180]: Done, updated

**Review by kaovilai — APPROVED** [review]
Looks sane.

**Review by sseago — APPROVED** [review]

**Comment by kaovilai** [comment]
woot woot

**Pull request merged**